conans/test/functional/toolchains/meson/test_meson_and_gnu_deps_flags.py

```diff
@@ -0,0 +1,61 @@
+import textwrap
+
+from conans.test.assets.sources import gen_function_cpp
+from conans.test.functional.toolchains.meson._base import TestMesonBase
+from conans.test.utils.tools import TestClient
+
+
+class TestMesonToolchainAndGnuFlags(TestMesonBase):
+
+    def test_mesontoolchain_using_gnu_deps_flags(self):
+        client = TestClient(path_with_spaces=False)
+        client.run("new hello/0.1 -s")
+        client.run("create . hello/0.1@ %s" % self._settings_str)
+        app = gen_function_cpp(name="main", includes=["hello"], calls=["hello"])
+
+        conanfile_py = textwrap.dedent("""
+        from conan import ConanFile
+        from conan.tools.meson import Meson, MesonToolchain
+        from conan.tools.gnu import get_gnu_deps_flags
+
+
+        class App(ConanFile):
+            settings = "os", "arch", "compiler", "build_type"
+            requires = "hello/0.1"
+
+            def layout(self):
+                self.folders.build = "build"
+
+            def generate(self):
+                # Get GNU flags from all the dependencies
+                cflags, cxxflags, cpp_flags, libs, ldflags = get_gnu_deps_flags(self)
```

Comment:
perhaps, we need some universal interface (object) that could be returned from or accepted by various tools (build helpers).
how about some `NamedTuple` or struct?